### ISSUE TITLE
CREATE TABLE AS with CASE

### ISSUE THREAD

**Issue by merlyn383** [dbeaver/dbeaver#5266]
#### System information: 
- Operating system (distribution) and version: Mac OS 10.14.2
- DBeaver version: Community Edition 5.3.4.201902031804
- Java version: 1.8.0_191-b12
- Additional extensions: Exasol Support 1.0.60.201902031804, DB2 Support 2.1.87.201902031804

#### Connection specification:
- Database name and version Oracle 12
- Driver name JDBC:oracle:thin
- Do you use tunnels or proxies (SSH, SOCKS, etc)? NO

#### Describe the problem you're observing:
When trying to use CREATE TABLE AS, if the SELECT statement has a CASE statement within it the statement/script bombs out with:
SQL Error [933] [42000]: ORA-00933: SQL command not properly ended

  ORA-00933: SQL command not properly ended

  ORA-00933: SQL command not properly ended

The same exact code can be run successfully in Oracle's SQLDeveloper

#### Steps to reproduce, if exist:

Running this statement fails in DBeaver:
CREATE TABLE temp01
AS 
(
	SELECT 
		CASE
			WHEN 1>1 THEN 1
			WHEN 1=2 THEN 2
			ELSE 3
		END AS "Example"
	FROM DUAL
);

Running this does not fail in DBeaver:
CREATE TABLE temp01
AS 
(
	SELECT 
		3 AS "Example"
	FROM DUAL
);
#### Include any warning/errors/backtraces from the logs

2019-02-12 13:14:34.813 - Error binding callable statement parameters from metadata: Unsupported feature
2019-02-12 13:14:34.814 - Error extracting parameters meta data: Unsupported feature
2019-02-12 13:14:34.998 - null
org.jkiss.dbeaver.DBException: SQL Error [933] [42000]: ORA-00933: SQL command not properly ended

	at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1665)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:100)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:101)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.jkiss.dbeaver.model.exec.DBCException: SQL Error [933] [42000]: ORA-00933: SQL command not properly ended

	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:840)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:2690)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:102)
	at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1633)
	... 3 more
Caused by: org.jkiss.dbeaver.DBException: SQL Error [933] [42000]: ORA-00933: SQL command not properly ended

	at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1665)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:400)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:835)
	... 6 more
Caused by: org.jkiss.dbeaver.model.exec.DBCException: SQL Error [933] [42000]: ORA-00933: SQL command not properly ended

	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:140)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:458)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:402)
	at org.jkiss.dbeaver.model.DBUtils.tryExecuteRecover(DBUtils.java:1633)
	... 8 more
Caused by: java.sql.SQLSyntaxErrorException: ORA-00933: SQL command not properly ended

	at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:450)
	at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:399)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1059)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:522)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:257)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:587)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:210)
	at oracle.jdbc.driver.T4CStatement.doOall8(T4CStatement.java:30)
	at oracle.jdbc.driver.T4CStatement.executeForRows(T4CStatement.java:931)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1150)
	at oracle.jdbc.driver.OracleStatement.executeInternal(OracleStatement.java:1792)
	at oracle.jdbc.driver.OracleStatement.execute(OracleStatement.java:1745)
	at oracle.jdbc.driver.OracleStatementWrapper.execute(OracleStatementWrapper.java:334)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:344)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:137)
	... 11 more

**Comment by mkgrgis** [comment]
There is no problem for PostgreSQL for
```
CREATE TABLE temp01 AS
( SELECT CASE
WHEN 1>1 THEN 1
WHEN 1=2 THEN 2
ELSE 3
END
AS "Example" );
```
It seems the problem is in Oracle jdbc structures. The point of problem is jdbc.driver.T4CStatement.executeForRows(T4CStatement.java:931)
I think its useful to send/escalate bugreport to oraclejdbc tracker.

**Comment by p91paul** [comment]
Oracle requires *some* statements to be sent to the database with a trailing ; while others need to be sent *without* the trailing ;. DBeaver tries to understand what kind of statement this is and remove the semicolon if the driver will have issues, but sometimes fails and removes semicolons where they are needed or vice-versa. As a workaround you can select the whole statement including ; and execute it. If it doesn't work, try to select the whole statement *excluding* ; and execute it.

**Comment by merlyn383** [comment]
I've been able to somewhat get around this by doing as was suggested, selecting the statement _without_ the semicolon and executing it but when I'm working collaboratively on code written by others it's a real pain.

Is there any chance of DBeaver addressing this in an update? If the statement is CREATE TABLE * AS and the driver is Oracle then drop off the semicolon?

**Comment by uslss** [comment]
fixed in scope of #7701 
Ticket closed
